### PR TITLE
Preparation for Water Evaporation and Salt Precipitation

### PR DESCRIPTION
**Pull request by goncalvesmachadoc**

### CODE REVIEW

**Comment by goncalvesmachadoc** [comment]
Hi @totto82 , could you review this code?

File: opm/material/fluidstates/BlackOilFluidState.hpp [hunk 1]
```diff
@@ -99,7 +99,9 @@ template <class ScalarT,
           bool enableTemperature = false,
           bool enableEnergy = false,
           bool enableDissolution = true,
+          bool enableEvaporation = false,
           bool enableBrine = false,
+          bool enableSolidSalt = false,
           unsigned numStoragePhases = FluidSystem::numPhases>
```
Comment (totto82) [inline opm/material/fluidstates/BlackOilFluidState.hpp:105]: Suggestion: Rename to enableSaltPrecipitation?

**Review by totto82 — CHANGES_REQUESTED** [review]
Thanks for the contribution. These changes looks good itself but needs to be merged together with changes in opm-common (https://github.com/OPM/opm-common/pull/1956) and opm-models/opm-simulators (no PR yet) since this changes the number of template parameters in the BlackoilFluidState.

**Comment by totto82** [comment]
jenkins build this please

**Comment by goncalvesmachadoc** [comment]
Updated the template parameters in https://github.com/OPM/opm-models/pull/678 and https://github.com/OPM/opm-simulators/pull/3715

**Comment by totto82** [comment]
Thanks for the update. The code change look ok from my perspective. Since these PRs cross repository, some more Jenkins testing will be needed. I will start the process. I assume there will be more PRs in opm-models and opm-simulators that implement the actual model so lets get these "Preparation" PRs merged!

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this opm-common=1956 opm-models=678 opm-simulators=3715 please

**Pull request merged**